### PR TITLE
adds prometheus metrics

### PR DESCRIPTION
**Pull request by bwalsh**
@buchanae  @kellrott @adamstruck  @prismofeverything : This dos_connect PR adds a  prometheus metrics endpoint `/metrics` and updates deployment documentation.

addresses:
https://github.com/ohsu-comp-bio/exastack/issues/14
https://github.com/ohsu-comp-bio/exastack/issues/18

e.g.
```
$ curl localhost:8080/metrics
# HELP python_info Python platform information
# TYPE python_info gauge
python_info{implementation="CPython",major="2",minor="7",patchlevel="13",version="2.7.13"} 1.0
# HELP dos_connect_data_bundles_count Number of data_bundles
# TYPE dos_connect_data_bundles_count gauge
dos_connect_data_bundles_count 100.0
# HELP dos_connect_data_objects_count Number of data_objects
# TYPE dos_connect_data_objects_count gauge
dos_connect_data_objects_count 4729.0
```

### CODE REVIEW

**Comment by buchanae** [comment]
Looks like you forgot to update requirements.txt

File: dos_connect/server/elasticsearch_backend.py [hunk 1]
```diff
@@ -126,3 +126,22 @@ def delete(properties, index='data_objects'):
...
+    return document counts
+    """
+    def _count(index):
+        try:
+            s = Search(using=client, index=index, doc_type=index[:-1])
+            return s.count()
+        except Exception as e:
+            log.error('error getting count of documents in {}'.format(index))
+            log.exception(e)
+            return 0
```
Comment (buchanae) [inline dos_connect/server/elasticsearch_backend.py]: https://prometheus.io/docs/instrumenting/writing_exporters/#failed-scrapes

Not sure, but I think you want to raise an error in this case, or perhaps don't emit a metric. I think by returning zero, you're hiding the error from Prometheus and telling it the count is legitimately zero.

**Comment by bwalsh** [comment]
@buchanae  thanks. done.

File: dos_connect/server/elasticsearch_backend.py (hunk 1 above)
Comment (bwalsh) [reply dos_connect/server/elasticsearch_backend.py]: Good point.  There is an edge case (no documents written yet to a blank db).  I've accounted for that.

**Pull request merged**